pkg/vgmanager/vgmanager_controller.go

```diff
@@ -243,3 +284,32 @@ func ToleratesTaints(tolerations []corev1.Toleration, taints []corev1.Taint) boo
 	}
 	return true
 }
+
+func (r *VGReconciler) updateStatus(ctx context.Context, status string, reason string, instance *lvmv1alpha1.LVMVolumeGroup) error {
+
+	found := false
+
+	nodeStatuses := instance.Status.NodeStatus
+	for _, nodeStatus := range nodeStatuses {
+		if nodeStatus.Node == r.Nodename {
+			found = true
+			nodeStatus.Status = status
+			nodeStatus.Reason = reason
+		}
+	}
+	if !found {
```

Comment:
I don't think it will update the `Status` when the `NodeName` was `found` but it has different `Status` or `Reason`.  So we should add status if NodeName not found and update status if NodeName is found but its `Status` or `Reason` are different.

Reply:
Didn't understand this.

Reply:
We are only updating the `instance.Status.NodeStatus` in line 307, inside the ` if !found` block.   What if `found` is `true` but `Status` and/or `Reason` is different. In this case, the `Status` and `Reason` will never be updated.

Reply:
The status is being updated in the original array if the node is found. append should create a new slice which is why it is being assigned again.

Reply:
got it.